enterprise/coderd/prebuilds/metricscollector_test.go

```diff
@@ -334,3 +451,33 @@ func findMetric(metricsFamilies []*prometheus_client.MetricFamily, name string,
 	}
 	return nil
 }
+
+// findAllMetricSeries finds all metrics with a given set of labels.
+func findAllMetricSeries(metricsFamilies []*prometheus_client.MetricFamily, labels map[string]string) map[string]*prometheus_client.Metric {
+	series := make(map[string]*prometheus_client.Metric)
+	for _, metricFamily := range metricsFamilies {
+		for _, metric := range metricFamily.GetMetric() {
+			labelPairs := metric.GetLabel()
+
+			if len(labelPairs) != len(labels) {
+				continue
+			}
+
+			// Convert label pairs to map for easier lookup
+			metricLabels := make(map[string]string, len(labelPairs))
+			for _, label := range labelPairs {
+				metricLabels[label.GetName()] = label.GetValue()
+			}
+
+			// Check if all requested labels match
+			for wantName, wantValue := range labels {
```

Comment:
this does the match one way, but not bidirectionally. we might get additional labels in metricLabels that this doesn't check for. I don't think that that is likely in this test at all. It might be fine, but if there's a quick fix like a compare function that will do this for you that might be something to consider.

Reply:
See above:

```
if len(labelPairs) != len(labels) {
  continue
}
```

Reply:
Shame there's no facepalm reaction here. You're right. Solid!